core/lib/brightcontent/base_controller_ext/fields.rb

```diff
@@ -24,6 +25,20 @@ def filter_fields
         []
       end
 
+      # Returns filter field definitions as a nested array.
+      #
+      #  [:foo, { bar: { as: :select }, qux: { as: :string } }]
+      #
+      # Becomes:
+      #
+      #  [[:foo], [:bar, { as: :select }], [:qux, { as: :string }]]
+      #
+      def filter_fields_with_options
+        filter_fields.map do |field, result|
+          field.is_a?(Hash) ? field.to_a : [[field]]
+        end.flatten(1)
+      end
```

Comment:
> Perhaps this belongs in a helper method?

Makes sense. Will do.

> Also, perhaps a better name for this is normalize_filter_fields? Because that's what it does, right?

I actually considered that too (or `normalized_filter_fields` to express that it's query method and not a command). I don't prefer either one, so I'd be happy to change it.

> And you should use flat_map.

Wow, 9 yrs and never heard of that method. I should have known Ruby always has a method for it. Thanks for mentioning it.